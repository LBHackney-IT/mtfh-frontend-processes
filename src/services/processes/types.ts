export enum Trigger {
  CloseProcess = "CloseProcess",
  CancelProcess = "CancelProcess",
}
