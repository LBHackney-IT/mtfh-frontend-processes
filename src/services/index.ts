export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
