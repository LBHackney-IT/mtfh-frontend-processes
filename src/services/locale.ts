import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
