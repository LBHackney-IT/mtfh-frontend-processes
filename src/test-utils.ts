import {
  getAssetV1,
  getPersonV1,
  getProcessV1,
  getTenureV1,
  mockProcessV1,
  patchProcessV1,
  postProcessV1,
  server,
} from "@hackney/mtfh-test-utils";

beforeEach(() => {
  server.use(getTenureV1());
  server.use(getPersonV1());
  server.use(getAssetV1());
  server.use(postProcessV1());
  server.use(getProcessV1());
  server.use(patchProcessV1());
});

export const mockProcessSelectTenants = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "SelectTenants" },
};

export const mockProcessAutomatedChecksFailed = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "AutomatedChecksFailed" },
};

export const mockProcessAutomatedChecksPassed = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "AutomatedChecksPassed" },
};

export const mockProcessInvalidState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "InvalidState" },
};

export const mockManualChecksPassedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "ManualChecksPassed" },
};

export const mockManualChecksFailedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "ManualChecksFailed" },
};

export const mockBreachChecksFailedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "BreachChecksFailed" },
};

export const mockBreachChecksPassedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "BreachChecksPassed" },
};

export const mockDocumentsRequestedDes = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "DocumentsRequestedDes" },
};

export const mockProcessClosedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "ProcessClosed" },
};

export const mockDocumentsRequestedDesAppointment = {
  ...mockProcessV1,
  currentState: {
    ...mockProcessV1.currentState,
    state: "DocumentsRequestedAppointment",
    processData: {
      formData: {
        appointmentDateTime: "2022-10-12T08:59:00.000Z",
      },
      documents: [],
    },
  },
  previousStates: [{ ...mockProcessV1.currentState, state: "DocumentsRequestedDes" }],
};

export const mockDocumentsRequestedAppointment = ({
  appointmentDateTime,
}: {
  appointmentDateTime: string;
}) => {
  return {
    ...mockProcessV1,
    currentState: {
      ...mockProcessV1.currentState,
      state: "DocumentsRequestedAppointment",
      processData: {
        formData: {
          appointmentDateTime,
        },
        documents: [],
      },
    },
  };
};

export const mockDocumentsAppointmentRescheduled = {
  ...mockProcessV1,
  currentState: {
    ...mockProcessV1.currentState,
    state: "DocumentsAppointmentRescheduled",
  },
};
