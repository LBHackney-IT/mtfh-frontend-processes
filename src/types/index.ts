export type TargetType = "tenure" | "property" | "person";

export interface IStartProcess {
  thirdPartyCondition?: string;
  thirdPartyComponent?: React.FC;
  subHeading?: string;
  subComponent?: React.FC;
}

export interface IProcess {
  processName: string;
  title: string;
  targetType: TargetType;
  startProcess: IStartProcess;
  states: {
    [state: string]: {
      state: string;
      triggers: Record<string, string>;
    };
  };
}
