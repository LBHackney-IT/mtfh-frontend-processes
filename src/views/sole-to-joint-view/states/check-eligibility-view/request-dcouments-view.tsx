import { Form, Formik } from "formik";

import { BulletWithExplanation } from "./shared";

import {
  splitContactDetailsByType,
  useContactDetails,
} from "@mtfh/common/lib/api/contact-details/v2";
import { editProcess } from "@mtfh/common/lib/api/process/v1";
import { HouseholdMember } from "@mtfh/common/lib/api/tenure/v1/types";
import {
  Button,
  Heading,
  InlineField,
  List,
  Radio,
  RadioGroup,
  Text,
} from "@mtfh/common/lib/components";

export interface RequestDcoumentsViewProps {
  tenant: HouseholdMember;
  process: any;
  processConfig: any;
  mutate: any;
}

export const RequestDcoumentsView = (props: RequestDcoumentsViewProps) => {
  const { tenant, process, processConfig, mutate } = props;
  const stateConfig = processConfig.states.breachChecksPassed;
  const { data: contacts } = useContactDetails(tenant.id);
  const { emails, phones } = splitContactDetailsByType(contacts?.results || []);

  return (
    <>
      <Heading variant="h3">Suporting documents</Heading>
      <Text size="sm">
        The following documentation is required from the secure tenant and/or proposed
        tenant, as proof to support their application:
      </Text>
      <List variant="bullets">
        <BulletWithExplanation
          text="Secure and Proposed tenant: Two forms of proof of identity for both the
              secure and proposed tenant. At least one each must be photographic ID"
          explanation="for example: valid passport, driving licence, bank statement, utility bill"
        />
        <BulletWithExplanation
          text="Proposed tenant: Proof of immigration status"
          explanation="for example: passport, home office letter, embassy letter, immigration status document"
        />
        <BulletWithExplanation
          text="Proposed tenant: Proof of relationship to the exisiting tenant"
          explanation="for example: marriage or civil partner certificate"
        />
        <BulletWithExplanation
          text="Proposed tenant: Proof of co-habitation: three documents proving 12 months
                residency at the property. If marriage certificate provided, any Proof of
                Address can be accepted."
          explanation="for example: letter, utility bill, council tax bill"
        />
      </List>
      <Heading variant="h3">Checking suporting documents</Heading>
      <Text size="sm">
        You must make an appointment with the tenant to check supporting documents
        in-person.
      </Text>
      <Text size="sm">{tenant.fullName} contact details:</Text>
      <Text size="sm">
        Phone:
        <span style={{ marginLeft: "1em" }}>{phones?.[0]?.contactInformation.value}</span>
      </Text>
      <Text size="sm">
        Email:
        <span style={{ marginLeft: "1em" }}>{emails?.[0]?.contactInformation.value}</span>
      </Text>
      <Formik
        initialValues={{ requestType: undefined }}
        onSubmit={async (values) => {
          let formData = {};
          if (values.requestType === "manual") {
            formData = {};
          }
          try {
            await editProcess({
              id: process.id,
              processTrigger: stateConfig.triggers.requestDocumentsDes,
              processName: process?.processName,
              etag: process.etag || "",
              formData,
              documents: [],
            });
            mutate();
          } catch (e: any) {
            console.log(e.response?.status || 500);
          }
        }}
      >
        {(props) => {
          const {
            values: { requestType },
          } = props;
          const buttonDisabled = requestType === undefined;
          return (
            <Form
              noValidate
              id="request-documents-form"
              className="request-documents-form"
            >
              <RadioGroup>
                <InlineField name="requestType" type="radio">
                  <Radio id="requestType-automatic" value="automatic">
                    Request documents electronically
                  </Radio>
                </InlineField>
                <InlineField name="requestType" type="radio">
                  <Radio id="requestType-manual" value="manual">
                    I have made an appointment to check supporting documents
                  </Radio>
                </InlineField>
              </RadioGroup>
              <Button type="submit" disabled={buttonDisabled}>
                Next
              </Button>
            </Form>
          );
        }}
      </Formik>
    </>
  );
};
