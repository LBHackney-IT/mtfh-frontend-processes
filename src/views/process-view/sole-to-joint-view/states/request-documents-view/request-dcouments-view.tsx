import { CheckSupportingDocuments } from "../../../../../components/check-supporting-documents";
import { locale } from "../../../../../services";
import { IProcess, ProcessComponentProps } from "../../../../../types";
import { BulletWithExplanation } from "../../../shared/process-components";
import { EligibilityChecksPassedBox } from "../shared";

import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Center,
  ErrorSummary,
  Heading,
  List,
  Spinner,
  Text,
} from "@mtfh/common/lib/components";
import { useErrorCodes } from "@mtfh/common/lib/hooks";

export interface RequestDocumentsViewProps extends ProcessComponentProps {
  processConfig: IProcess;
}

export const RequestDocumentsView = ({
  process,
  mutate,
  optional,
}: RequestDocumentsViewProps) => {
  const { closeProcessReason } = optional;
  const errorMessages = useErrorCodes();
  const { data: tenure, error } = useTenure(process.targetId);

  if (error) {
    return (
      <ErrorSummary
        id="request-documents-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure || !errorMessages) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  return (
    <div data-testid="soletojoint-RequestDocuments">
      <EligibilityChecksPassedBox />
      {!closeProcessReason && (
        <>
          <Heading variant="h3">{locale.supportingDocuments}</Heading>
          <Text>
            The following documentation is required from the secure tenant and/or proposed
            tenant, as proof to support their application:
          </Text>
          <List variant="bullets">
            <BulletWithExplanation
              text="Secure and Proposed tenant: Two forms of proof of identity for both the
              secure and proposed tenant. At least one each must be photographic ID"
              explanation="for example: valid passport, driving licence, bank statement, utility bill"
            />
            <BulletWithExplanation
              text="Proposed tenant: Proof of immigration status"
              explanation="for example: passport, home office letter, embassy letter, immigration status document"
            />
            <BulletWithExplanation
              text="Proposed tenant: Proof of relationship to the existing tenant"
              explanation="for example: marriage or civil partner certificate"
            />
            <BulletWithExplanation
              text="Proposed tenant: Proof of co-habitation: three documents proving 12 months
                residency at the property. If marriage certificate provided, any Proof of
                Address can be accepted."
              explanation="for example: letter, utility bill, council tax bill"
            />
          </List>

          <CheckSupportingDocuments
            process={process}
            mutate={mutate}
            optional={{ tenant, declaration: true }}
          />
        </>
      )}
    </div>
  );
};
