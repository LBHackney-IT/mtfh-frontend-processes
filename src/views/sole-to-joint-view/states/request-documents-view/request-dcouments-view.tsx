import { Form, Formik } from "formik";

import { SoleToJointHeader } from "../../../../components";
import { DateTimeFields } from "../../../../components/appointment-form/appointment-form";
import {
  RequestDocumentsFormData,
  requestDocumentsSchema,
} from "../../../../schemas/request-documents";
import { locale } from "../../../../services";
import { IProcess } from "../../../../types";
import { dateToString, stringToDate } from "../../../../utils/date";
import {
  BulletWithExplanation,
  EligibilityChecksPassedBox,
  TenantContactDetails,
} from "../shared";

import { Process, editProcess } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Button,
  Center,
  ErrorSummary,
  Heading,
  InlineField,
  List,
  Radio,
  RadioGroup,
  Spinner,
  Text,
} from "@mtfh/common/lib/components";
import { useErrorCodes } from "@mtfh/common/lib/hooks";

export interface RequestDocumentsViewProps {
  process: Process;
  processConfig: IProcess;
  mutate: any;
}

export const RequestDocumentsView = ({
  process,
  processConfig,
  mutate,
}: RequestDocumentsViewProps) => {
  const errorMessages = useErrorCodes();
  const { data: tenure, error } = useTenure(process.targetId);

  if (error) {
    return (
      <ErrorSummary
        id="request-documents-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure || !errorMessages) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const stateConfig = processConfig.states.breachChecksPassed;
  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  return (
    <div data-testid="soletojoint-RequestDocuments">
      <SoleToJointHeader processConfig={processConfig} process={process} />
      <EligibilityChecksPassedBox />
      <Heading variant="h3">{locale.supportingDocuments}</Heading>
      <Text size="sm">
        The following documentation is required from the secure tenant and/or proposed
        tenant, as proof to support their application:
      </Text>
      <List variant="bullets">
        <BulletWithExplanation
          text="Secure and Proposed tenant: Two forms of proof of identity for both the
              secure and proposed tenant. At least one each must be photographic ID"
          explanation="for example: valid passport, driving licence, bank statement, utility bill"
        />
        <BulletWithExplanation
          text="Proposed tenant: Proof of immigration status"
          explanation="for example: passport, home office letter, embassy letter, immigration status document"
        />
        <BulletWithExplanation
          text="Proposed tenant: Proof of relationship to the existing tenant"
          explanation="for example: marriage or civil partner certificate"
        />
        <BulletWithExplanation
          text="Proposed tenant: Proof of co-habitation: three documents proving 12 months
                residency at the property. If marriage certificate provided, any Proof of
                Address can be accepted."
          explanation="for example: letter, utility bill, council tax bill"
        />
      </List>
      <Heading variant="h3">Checking supporting documents</Heading>
      <Text size="sm">
        You can request supporting documents through the Document Evidence Store or you
        can make an appointment with the tenant to check supporting documents in-person.
      </Text>
      {tenant ? <TenantContactDetails tenant={tenant} /> : <Text>Tenant not found.</Text>}
      <Formik<RequestDocumentsFormData>
        initialValues={{
          requestType: "",
          day: "",
          month: "",
          year: "",
          hour: "",
          minute: "",
          amPm: "",
        }}
        validateOnBlur={false}
        validateOnChange={false}
        validationSchema={requestDocumentsSchema(errorMessages)}
        onSubmit={async (values) => {
          const { requestType, day, month, year, hour, minute, amPm } = values;
          let processTrigger = stateConfig.triggers.requestDocumentsDes;
          let formData = {};
          if (requestType === "manual") {
            processTrigger = stateConfig.triggers.requestDocumentsAppointment;
            const date = stringToDate(
              `${day}/${month}/${year} ${hour}:${minute} ${amPm.toUpperCase()}`,
              "dd/MM/yyyy hh:mm a",
            );
            formData = {
              appointmentDateTime: dateToString(date, "yyyy-MM-dd'T'HH:mm:ss"),
            };
          }
          try {
            await editProcess({
              id: process.id,
              processName: process?.processName,
              etag: process.etag || "",
              processTrigger,
              formData,
              documents: [],
            });
            mutate();
          } catch (e: any) {
            console.log(e.response?.status || 500);
          }
        }}
      >
        {(props) => {
          const {
            values,
            values: { requestType },
            setFieldValue,
          } = props;
          return (
            <Form
              noValidate
              id="request-documents-form"
              className="request-documents-form"
            >
              <RadioGroup>
                <InlineField name="requestType" type="radio">
                  <Radio
                    id="requestType-automatic"
                    value="automatic"
                    onClick={() => {
                      if (requestType === "manual") {
                        setFieldValue("day", "01");
                        setFieldValue("month", "01");
                        setFieldValue("year", "3000");
                        setFieldValue("hour", "01");
                        setFieldValue("minute", "01");
                        setFieldValue("amPm", "am");
                      }
                    }}
                  >
                    Request documents electronically
                  </Radio>
                </InlineField>
                <InlineField name="requestType" type="radio">
                  <Radio
                    id="requestType-manual"
                    value="manual"
                    onClick={() => {
                      if (requestType === "automatic") {
                        setFieldValue("day", "");
                        setFieldValue("month", "");
                        setFieldValue("year", "");
                        setFieldValue("hour", "");
                        setFieldValue("minute", "");
                        setFieldValue("amPm", "");
                      }
                    }}
                  >
                    I have made an appointment to check supporting documents
                  </Radio>
                </InlineField>
              </RadioGroup>
              {requestType === "manual" && <DateTimeFields />}
              <Button
                type="submit"
                disabled={
                  !Object.values(values).some((value) => {
                    return value !== "";
                  })
                }
              >
                Next
              </Button>
            </Form>
          );
        }}
      </Formik>
    </div>
  );
};
