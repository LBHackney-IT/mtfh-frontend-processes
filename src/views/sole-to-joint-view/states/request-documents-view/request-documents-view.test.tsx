import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { processes } from "../../../../services";
import { mockBreachChecksPassedState } from "../../../../test-utils";
import { RequestDcoumentsView } from "./request-dcouments-view";

import * as contactDetailsService from "@mtfh/common/lib/api/contact-details/v2/service";
import { ContactDetailsResponse } from "@mtfh/common/lib/api/contact-details/v2/service";
import {
  ContactDetail,
  ContactInformationContactTypes,
} from "@mtfh/common/lib/api/contact-details/v2/types";
import { HouseholdMember } from "@mtfh/common/lib/api/tenure/v1/types";
import { AxiosSWRResponse } from "@mtfh/common/lib/http";

describe("request-documents-view", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  const tenant = { id: "tenant-id" } as HouseholdMember;
  const contactDetailsResponse = {
    results: [
      {
        contactInformation: {
          contactType: ContactInformationContactTypes.EMAIL,
          value: "testtest34335@gmail.com",
        },
      } as ContactDetail,
      {
        contactInformation: {
          contactType: ContactInformationContactTypes.PHONE,
          value: "+44(0)7700 900251",
        },
      } as ContactDetail,
    ],
  } as ContactDetailsResponse;

  test("it renders RequestDocuments correctly on state BreachChecksPassed", async () => {
    jest.spyOn(contactDetailsService, "useContactDetails").mockReturnValue({
      data: contactDetailsResponse,
    } as AxiosSWRResponse<ContactDetailsResponse>);

    const { container } = render(
      <RequestDcoumentsView
        processConfig={processes.soletojoint}
        process={mockBreachChecksPassedState}
        mutate={() => {}}
        tenant={tenant}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    expect(container).toMatchSnapshot();
  });

  test("it enables date/time option and next button", async () => {
    const user = userEvent.setup();

    jest.spyOn(contactDetailsService, "useContactDetails").mockReturnValue({
      data: contactDetailsResponse,
    } as AxiosSWRResponse<ContactDetailsResponse>);

    const { container } = render(
      <RequestDcoumentsView
        processConfig={processes.soletojoint}
        process={mockBreachChecksPassedState}
        mutate={() => {}}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );

    const radioButton = await screen.findByText(
      "I have made an appointment to check supporting documents",
    );
    await user.click(radioButton);
    await user.type(screen.getByPlaceholderText(/dd/i), "01");
    await user.type(screen.getByPlaceholderText(/mm/i), "01");
    await user.type(screen.getByPlaceholderText(/yy/i), "2022");
    await user.type(screen.getAllByPlaceholderText(/00/i)[0], "01");
    await user.type(screen.getAllByPlaceholderText(/00/i)[1], "01");
    await user.type(screen.getByPlaceholderText(/am/i), "am");

    expect(container).toMatchSnapshot();
  });
});
