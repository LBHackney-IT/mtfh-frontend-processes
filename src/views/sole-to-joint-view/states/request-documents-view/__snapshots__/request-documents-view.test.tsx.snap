// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request-documents-view it enables date/time option and next button 1`] = `
<div>
  <div
    data-testid="soletojoint-RequestDocuments"
  >
    <h1
      class="lbh-heading-h1"
    >
      Sole tenant requests a joint tenure
    </h1>
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <h3
      class="lbh-heading-h3"
    >
      Supporting documents
    </h3>
    <p
      class="lbh-body-s"
    >
      The following documentation is required from the secure tenant and/or proposed tenant, as proof to support their application:
    </p>
    <ul
      class="lbh-list lbh-list--bullet"
    >
      <li>
        <p
          class="lbh-body-s"
          style="font-weight: bold;"
        >
          Secure and Proposed tenant: Two forms of proof of identity for both the secure and proposed tenant. At least one each must be photographic ID
        </p>
        <span
          class="lbh-body-s"
        >
          (
          for example: valid passport, driving licence, bank statement, utility bill
          )
        </span>
      </li>
      <li>
        <p
          class="lbh-body-s"
          style="font-weight: bold;"
        >
          Proposed tenant: Proof of immigration status
        </p>
        <span
          class="lbh-body-s"
        >
          (
          for example: passport, home office letter, embassy letter, immigration status document
          )
        </span>
      </li>
      <li>
        <p
          class="lbh-body-s"
          style="font-weight: bold;"
        >
          Proposed tenant: Proof of relationship to the existing tenant
        </p>
        <span
          class="lbh-body-s"
        >
          (
          for example: marriage or civil partner certificate
          )
        </span>
      </li>
      <li>
        <p
          class="lbh-body-s"
          style="font-weight: bold;"
        >
          Proposed tenant: Proof of co-habitation: three documents proving 12 months residency at the property. If marriage certificate provided, any Proof of Address can be accepted.
        </p>
        <span
          class="lbh-body-s"
        >
          (
          for example: letter, utility bill, council tax bill
          )
        </span>
      </li>
    </ul>
    <h3
      class="lbh-heading-h3"
    >
      Checking supporting documents
    </h3>
    <p
      class="lbh-body-s"
    >
      You must make an appointment with the tenant to check supporting documents in-person.
    </p>
    <p
      class="lbh-body-s"
    >
      Global Web Consultant Alverta Denesik
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        +44(0)7700 900251
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        testtest34335@gmail.com
      </span>
    </p>
    <form
      action="#"
      class="request-documents-form"
      id="request-documents-form"
      novalidate=""
    >
      <div
        class="govuk-radios lbh-radios"
      >
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="requestType-automatic"
            name="requestType"
            type="radio"
            value="automatic"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="requestType-automatic"
          >
            Request documents electronically
          </label>
        </div>
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="requestType-manual"
            name="requestType"
            type="radio"
            value="manual"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="requestType-manual"
          >
            I have made an appointment to check supporting documents
          </label>
        </div>
      </div>
      <div
        style="display: flex;"
      >
        <fieldset
          class="govuk-form-group lbh-form-group mtfh-appointment-form__date"
          id="appointment-form-date"
        >
          <legend
            class="govuk-label lbh-label"
            for="appointment-form-date-field"
          >
            Date
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </legend>
          <div
            class="govuk-date-input lbh-date-input"
          >
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-date-field-day"
              >
                Day
              </label>
              <input
                aria-label="Day"
                aria-valuemax="31"
                aria-valuemin="1"
                aria-valuenow="1"
                aria-valuetext="01"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="day"
                placeholder="dd"
                required=""
                role="spinbutton"
                value="01"
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-date-field-month"
              >
                Month
              </label>
              <input
                aria-label="Month"
                aria-valuemax="12"
                aria-valuemin="1"
                aria-valuenow="1"
                aria-valuetext="01"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="month"
                placeholder="mm"
                required=""
                role="spinbutton"
                value="01"
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-date-field-year"
              >
                Year
              </label>
              <input
                aria-label="Year"
                aria-valuenow="2022"
                aria-valuetext="2022"
                class="govuk-input lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                name="year"
                placeholder="yyyy"
                required=""
                role="spinbutton"
                value="2022"
              />
            </div>
          </div>
        </fieldset>
        <fieldset
          class="govuk-form-group lbh-form-group mtfh-appointment-form__time"
          id="appointment-form-time"
          style="margin-top: 0px; margin-left: 5em;"
        >
          <legend
            class="govuk-label lbh-label"
            for="appointment-form-time-field"
          >
            Time
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </legend>
          <div
            class="govuk-date-input lbh-date-input"
          >
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-time-field-hour"
              >
                Hour
              </label>
              <input
                aria-label="Hour"
                aria-valuemax="12"
                aria-valuemin="0"
                aria-valuenow="1"
                aria-valuetext="01"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="hour"
                placeholder="00"
                required=""
                role="spinbutton"
                value="01"
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-time-field-minute"
              >
                Minute
              </label>
              <input
                aria-label="Minute"
                aria-valuemax="59"
                aria-valuemin="0"
                aria-valuenow="1"
                aria-valuetext="01"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="minute"
                placeholder="00"
                required=""
                role="spinbutton"
                value="01"
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-time-field-amPm"
              >
                am/pm
              </label>
              <input
                aria-label="AM/PM"
                class="govuk-input lbh-input govuk-input--width-2"
                name="amPm"
                placeholder="am"
                value="AMam"
              />
            </div>
          </div>
        </fieldset>
      </div>
      <button
        class="govuk-button lbh-button"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</div>
`;

exports[`request-documents-view it renders RequestDocuments correctly on state BreachChecksPassed 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
  >
    <svg
      class="mtfh-icon"
      focusable="false"
      height="50"
      stroke="#00703c"
      viewBox="0 0 42 42"
      width="50"
    >
      <title>
        Loading...
      </title>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(3 3)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            transform="rotate(112.708 18 18)"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
